### DISCUSSION TITLE
inner join

### DISCUSSION THREAD

**Discussion by peterkimzz** [supabase/supabase#2438] [Questions] [answered]
Here is my code.

```js
const { data: campaigns, error } = await supabase
      .from("campaigns")
      .select( `
        *,
        user_join_campaigns (*)
      `)
      .filter("user_join_campaigns.user_id", "eq", user_id);
```

The result shows me all of records regardless of `filter`.

How to pick the `campaigns` record only matched by `user_id`?

**Comment by steve-chavez** [comment, marked as answer]
Hey @peterkimzz,

The inner join is a future enhancement: https://github.com/supabase/postgrest-js/issues/197. For now the workaround would be filtering client side.

**Comment by r100-stack** [comment]
Another work around is to use a view. You can create a view using a custom sql statement that does the inner join, exactly like how we want it. Then using the same supabase library on the client side, we can just filter from that view, instead of the joined table.

Example
```python
Table: group(id, name)
1, Group A
2, Group B
3, Group C

Table: group_members(group_id, user_id)
1, Alison
1, Bob
3, Alison
2, Henry
```

## Goal: To display all the groups Alison is in

if we use the below pseudocode,

```javascript
supabase.from('group').select(*, group_members).eq('user_id', 'Alison')
```

then we get something like,

```json
{"id": 1, "group_members": ["Alison"]},
{"id": 2, "group_members": []},
{"id": 3, "group_members": ["Alison"]}
```

We still get the record from group 2, even though Alison is not in group 2.
Instead, we can make a view

```sql
create view group_view as select * from group, group_members where group.id = group_members.id;
```

Then from the supabase's client side, you can filter this view instead

```javascript
supabase.from('group_view').select(*).eq('user_id', 'Alison')
```

This will give something like

```json
{"id": 1, "user_id": "Alison"},
{"id": 3, "user_id": "Alison"}
```
